### ISSUE TITLE
serveStaticFiles cannot handle non ASCII file names on windows

### ISSUE THREAD

**Issue by aka-demik** [rejectedsoftware/vibe.d#1050]
It works fine with: `"subConfigurations": {"vibe-d": "win32"}`. So, we have a workaround.
But with `"subConfigurations": {"vibe-d": "libevent"}` I got 404 and this:
```
[01EF0000:01EF4600 dbg] Path 'Sehenswürdigkeit.txt' -> 'Sehenswürdigkeit.txt'
[01EF0000:01EF4600 dbg] Failed to open file public\Sehenswürdigkeit.txt: object.Exception@C:\Users\Denis.Hlyakin\AppData\Roaming\dub\packages\vibe-d-0.7.23\source\vibe\core\drivers\threadedfile.d(94): Failed to open file 'public\Sehenswürdigkeit.txt'.
...
[01EF0000:01EF4600 dbg] no route match: GET /Sehensw%C3%BCrdigkeit.txt
[01EF0000:01EF4600 dia] No response written for /Sehensw%C3%BCrdigkeit.txt
```
I think this is due to an old Phobos issue [7648](https://issues.dlang.org/show_bug.cgi?id=7648)

And with `"subConfigurations": {"vibe-d": "libasync"}` I got OK(200), empty answer and this:
```
[02280000:02284700 dbg] Path 'Sehenswurdigkeit.txt' -> 'Sehenswurdigkeit.txt'
[02280000:02284700 dia] Error while writing the response: Error in READ, std.exception.ErrnoException@std\stdio.d(372): Cannot open file `public\Sehenswurdigkeit.txt' in mode `rb' (No error)
[02280000:02284700 dia] ----------------
...
[02280000:02284700 dbg] Exception while handling request GET /Sehensw%C3%BCrdigkeit.txt: object.Exception@C:\Users\Denis.Hlyakin\AppData\Roaming\dub\packages\vibe-d-0.7.23\source\vibe\core\drivers\libasync.d(573): Error in READ, std.exception.ErrnoException@std\stdio.d(372): Cannot open file `public\Sehenswurdigkeit.txt' in mode `rb' (No error)
[02280000:02284700 dbg] ----------------
...
[02280000:02284700 dbg] ----------------
...
[02280000:02284700 dbg] HTTP response only written partially before finalization. Terminating connection.
```
Perhaps it is also a separate issue for libAsync. Since it is not correct to return OK after read errors.
I skipped the stack trace, since it contains only addresses without the module names and line numbers.
My test case:
```
import vibe.appmain;
import vibe.http.fileserver;
import vibe.http.router;
import vibe.http.server;

import std.file;
import std.path;

shared static this() {
	write("./public/Sehenswürdigkeit.txt", "https://issues.dlang.org/show_bug.cgi?id=7648");

	auto settings = new HTTPServerSettings;
	settings.port = 8080;
	settings.bindAddresses = ["::1", "127.0.0.1"];
	
	auto router = new URLRouter;
	router.get("/", staticRedirect("/Sehenswürdigkeit.txt"));
	router.get("*", serveStaticFiles("./public/",));
	
	listenHTTP(settings, router);
}
```

**Comment by aka-demik** [comment]
I made a [PR](https://github.com/D-Programming-Language/phobos/pull/3138) for phobos, that would fix it.
But can be as long as Issue 7648 is not closed, print a compile time warning on windows (if user call serveStaticFiles).
Since I noticed the problem only after deploying of the application.

**Comment by aka-demik** [comment]
Fixed in dmd-2.068.0-b1
